src/slot.c

```diff
@@ -47,13 +47,19 @@ static inline void node_list_init()
     conf_node_dec_ref(node);
 }
 
-static inline void node_list_add(struct node_info *node)
+static inline void node_list_add(struct address *dst, size_t *len, struct node_info *node)
```

Comment:
`node_list_*` are targeted at the global `node_list`, I guess we should keep the consistency somehow.